### PR TITLE
binding against incode dependency of lifecycle manager & various CLI alpha command improvements

### PR DESCRIPTION
**Pull request by jakobmoellersap**
**Description**

Changes proposed in this pull request:

- adds lifecycle manager as code dependency to CLI to allow typed API access
- changes interface of Kyma CLI to make use of SSA resource infos
- Completely removes binary dependency of kustomize and uses inbuilt version that is derived from the lifecycle manager dependency ensuring compatiblity
- Bumps all APIs to v1beta1 where applicable
- Changes kyma alpha deploy to add a flag called `lifecycle-manager` which can be used to override the lifecycle manager image that is deployed in the default kustomisation. can be used to dynamically override the lifecycle-manager while not having to download it.
- Changes kyma alpha enable/disable module to use a native controller-runtime watch that allows it to react much better to when the Kyma Custom Resource is updated
- Adds Steps to kyma enable/disable module
- Adds a time to all Kyma Commands printed by steps to see how long each step took without having to enable verbose logs
- Fixes Verbose mode for enable/disable module command as logger was not used correctly
- Fixes klog redirects that caused log messages to leak from Kubernetes libraries incorrectly, now they are redirected to the zap logger which can be enabled with `verbose`

**Related issue(s)**

### CODE REVIEW

**Review by ruanxin — DISMISSED** [review]

File: docs/gen-docs/kyma_alpha_verify_module.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+## Flags
+
+```bash
+  -c, --credentials string      Basic authentication credentials for the given registry in the format user:password
+      --insecure                Use an insecure connection to access the registry.
+      --key string              Specifies the path where the private key used for signing
+      --name string             Name of the module
+      --name-mapping string     Overrides the OCM Component Name Mapping, one of: "urlPath" or "sha256-digest" (default "urlPath")
```
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_verify_module.md:27]:
```suggestion
      --name-mapping string     Overrides the OCM Component Name Mapping. Use "urlPath" or "sha256-digest". (default "urlPath")
```

File: docs/gen-docs/kyma_alpha_verify_module.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Verifies the signature of a Kyma module bundled as an OCI container image.
+
+## Synopsis
+
+Use this command to verify a Kyma module.
+
+### Detailed description
+
+Kyma modules can be cryptographically signed to make sure they are correct and distributed by a trusted authority. This command verifies the authenticity of a given module.
```
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_verify_module.md:13]:
```suggestion
Kyma modules can be cryptographically signed to ensure they are correct and distributed by a trusted authority. This command verifies the authenticity of a given module.
```

File: docs/gen-docs/kyma_alpha_verify_module.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+
+```bash
+kyma alpha verify module --name MODULE_NAME --version MODULE_VERSION --registry MODULE_REGISTRY [flags]
+```
+
+## Flags
+
+```bash
+  -c, --credentials string      Basic authentication credentials for the given registry in the format user:password
```
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_verify_module.md:23]:
```suggestion
  -c, --credentials string      Basic authentication credentials for the given registry in the user:password format.
```

File: docs/gen-docs/kyma_alpha_verify_module.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      --insecure                Use an insecure connection to access the registry.
+      --key string              Specifies the path where the private key used for signing
```
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_verify_module.md:25]:
```suggestion
      --key string              Specifies the path where the private key is used for signing.
```

File: docs/gen-docs/kyma_alpha_verify_module.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      --name string             Name of the module
```
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_verify_module.md:26]:
```suggestion
      --name string             Name of the module.
```

File: docs/gen-docs/kyma_alpha_sign_module.md [hunk 6]
```diff
@@ -14,21 +14,21 @@ This command signs all module resources recursively based on an unsigned compone
...
-  -c, --credentials string       Basic authentication credentials for the given registry in the format user:password
-      --insecure                 Use an insecure connection to access the registry.
-      --mod-path string          Specifies the path where the signed component descriptor will be stored (default "./mod")
-      --nameMapping string       Overrides the OCM Component Name Mapping, one of: "urlPath" or "sha256-digest" (default "urlPath")
-      --private-key string       Specifies the path where the private key used for signing
-      --registry string          Repository context url where unsigned component descriptor located
-      --signature-name string    name of the signature for signing
-      --signed-registry string   Repository context url where signed component descriptor located
-  -t, --token string             Authentication token for the given registry (alternative to basic authentication).
+  -c, --credentials string      Basic authentication credentials for the given registry in the format user:password
```
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_sign_module.md:23]:
```suggestion
  -c, --credentials string      Basic authentication credentials for the given registry in the user:password format.
```

File: docs/gen-docs/kyma_alpha_deploy.md [hunk 7]
```diff
@@ -28,20 +27,24 @@ kyma alpha deploy [flags]
       --cert-manager string         Installs cert-manager from the specified static version. an empty string skips the installation. (default "v1.11.0")
   -c, --channel string              Select which channel to deploy from. (default "regular")
       --dry-run                     Renders the Kubernetes manifests without actually applying them.
-  -k, --kustomization stringArray   Provide one or more kustomizations to deploy. Each occurrence of the flag accepts a URL with an optional reference (commit, branch, or release) in the format URL@ref or a local path to the directory of the kustomization file.
-                                    	Defaults to deploying Lifecycle Manager and Module Manager from GitHub main branch.
-                                    	 (default [https://github.com/kyma-project/lifecycle-manager/config/default])
+  -f, --force-conflicts             Forces the patching of Kyma spec modules in case their managed field was edited by a source other than Kyma CLI.
+  -k, --kustomization stringArray   Provide one or more kustomizations to deploy.
```
[the remark this answers is not in the document]
Comment (jakobmoellersap) [reply docs/gen-docs/kyma_alpha_deploy.md:31]: `kustomize` is a tool so I will leave it with a `k` but I will change Provides

File: docs/gen-docs/kyma_alpha_create_module.md [hunk 8]
```diff
@@ -45,23 +45,26 @@ Build module my-domain/modB in version 3.2.1 and push it to a local registry "un
...
-  -r, --resource stringArray            Add an extra resource in a new layer with format <NAME:TYPE@PATH>. It is also possible to provide only a path; name will default to the last path element and type to 'helm-chart'
-      --sec-scan-cfg string             Path to the directory holding the security scan configuration file.
-  -t, --token string                    Authentication token for the given registry (alternative to basic authentication).
-      --version string                  Version of the module. This flag is mandatory.
+      --channel string                     Channel to use for the module template. (default "regular")
+  -c, --credentials string                 Basic authentication credentials for the given registry in the format user:password
+      --default-cr string                  File containing the default custom resource of the module. If the module is a kubebuilder project, the default CR will be automatically detected.
+      --descriptor-version string          Schema version to use for the generated OCM descriptor. One of ocm.software/v3alpha1,v2 (default "v2")
```
[the remark this answers is not in the document]
Comment (jakobmoellersap) [reply docs/gen-docs/kyma_alpha_create_module.md:51]: This one is correct, unfortunately a backwards compatible issue leads to one being `v2` and the other one being `ocm.software/v3alpha1`

**Review by lindnerby — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**